test/jdk/javax/swing/text/CSSBorder/6796710/bug6796710.java

```diff
@@ -140,6 +142,8 @@ private static BufferedImage getPnBottomImage() {
 
         Util.convertRectToScreen(rect, pnBottom.getParent());
 
-        return robot.createScreenCapture(rect);
+        MultiResolutionImage img = robot.createMultiResolutionScreenCapture(rect);
+        return (BufferedImage)img.getResolutionVariant(rect.width, rect.height);
+        //return robot.createScreenCapture(rect);
```

Comment:
Isn't the old and new code should get the same result? The old createScreenCapture always creates an image of the low dpi quality. The createMultiResolutionScreenCapture creates two images low/hi dpi, but since you request the same size "rect" the low dpi image is returned, no?

Reply:
Maybe it's different in iMac systems..I dont have access to this system to analyse further nor do I have much knowledge on this 2D/awt multiresolution image side of things..

Reply:
If I remember correctly, MRI will contain only one image captured by Robot, so it would return the only image it stores, which would not be scaled down if High DPI is in effect.

Reply:
On the HiDPI screen that MRI will have two images, and my expectation is that the image requested by the updated test and the image used before the fix should be the same.

Reply:
How to find those 2 bufferedimages returned by `createMultiResolutionScreenCapture `and `createScreenCapture `are same or not apart from pixel check (which shows obviously they are not same and color differ by 1)?

Reply:
Probably we can apply the defensive fix of color tolerance as per 1st iteration ?

Reply:
It still fails once or twice with createMultiResolutionScreenCapture if the test iteration is increased, so I think color tolerance, even if defensive, is the one to go for.

Reply:
> On the HiDPI screen that MRI will have two images, and my expectation is that the image requested by the updated test and the image used before the fix should be the same.

In this case, yes, the image will be the same as before: a scaled down bitmap you get with regular `createScreenCapture`.

> How to find those 2 bufferedimages returned by createMultiResolutionScreenCapture and createScreenCapture are same or not apart from pixel check (which shows obviously they are not same and color differ by 1)?

You can get the largest variant, and compare colors there. Yet the size of the two images will be different, most likely.

Reply:
> It still fails once or twice with createMultiResolutionScreenCapture if the test iteration is increased, so I think color tolerance, even if defensive, is the one to go for.

So we return back to the initial question, why the code which capture same window rendered in same color in the same location produce different results? Why the tolerance is needed, Is it possible that the robot or rendering generate some garbage or override/read something wrong in memory?